src/main/java/org/usfirst/frc4048/common/diag/DiagColorSensor.java

```diff
@@ -21,39 +25,35 @@ public class DiagColorSensor implements Diagnosable {
 
     private ColorSensor colorsensor;
     private String name;
-    private NetworkTableEntry networkTableEntry;
-    private Map<ColorSensor.ColorValue, Boolean> colorMap;
+    private String title;
+    private GenericEntry networkTableEntry;
+    private Map<ColorValue, Boolean> colorMap;
 
-    public DiagColorSensor(String name, ColorSensor colorsensor) {
+    public DiagColorSensor(String name, String title, ColorSensor colorsensor) {
         this.name = name;
+        this.title = title;
         this.colorsensor = colorsensor;
-        colorMap = new HashMap<ColorSensor.ColorValue, Boolean>();
+        colorMap = new HashMap<>(Arrays.stream(ColorValue.values()).map(colorValue -> new HashMap.SimpleEntry<>(colorValue, false)).collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue)));
```

Comment:
`.stream().collect(Collectors.toMap(colorValue -> Functions.identity, colorValue -> false))`

Reply:
Or, can also move that into the `reset()` call